subsys/net/lib/nrf_cloud/src/nrf_cloud_agps.c

```diff
@@ -7,6 +7,7 @@
 #include <zephyr.h>
 #include <device.h>
 #include <drivers/gps.h>
+#include <nrf_modem_gnss.h>
```

Comment:
A general comment about changes to nRFCloud A-GPS. It's probably too late for these changes to go into NCS 1.6.0 anymore, but I think we still need to consider when we can remove support for GNSS socket and GPS driver. The GNSS socket is deprecated in NCS 1.6.0, but the GPS driver is not. So, we probably should support GPS driver also in NCS 1.7.0.

Supporting all the different APIs makes things more complicated, but I'm wondering if it would be possible to make this work with all three APIs. If there's no `gps_dev` and `fd == -1`, the new GNSS API would be used. What do you think?

Reply:
Well, if GPS Driver is to be kept, I would then rather see a fix in nrf_cloud that, when APP drops GPS Driver completely, nrf_cloud could use GPS Driver seamlessly write AGPS data to modem. Current failed with -9.